include/libopencm3/msp432/e4/gpio.h

```diff
@@ -0,0 +1,2524 @@
+
+#include <libopencm3/cm3/common.h>
+#include <libopencm3/msp432/e4/memorymap.h>
+#include <stdbool.h>
+
+/** @defgroup gpio_reg_base GPIO Register Base Addresses
+ * @brief GPIO Register Base Addresses
+@{*/
+/** GPIOA Base Address */
+#define GPIOA               GPIOA_BASE
+/** GPIOB Base Address */
+#define GPIOB               GPIOB_BASE
+/** GPIOC Base Address */
+#define GPIOC               GPIOC_BASE
+/** GPIOD Base Address */
+#define GPIOD               GPIOD_BASE
+/** GPIOE Base Address */
+#define GPIOE               GPIOE_BASE
+/** GPIOF Base Address */
+#define GPIOF               GPIOF_BASE
+/** GPIOG Base Address */
+#define GPIOG               GPIOG_BASE
+/** GPIOH Base Address */
+#define GPIOH               GPIOH_BASE
+/** GPIOJ Base Address */
+#define GPIOJ               GPIOJ_BASE
+/** GPIOK Base Address */
+#define GPIOK               GPIOK_BASE
+/** GPIOL Base Address */
+#define GPIOL               GPIOL_BASE
+/** GPIOM Base Address */
+#define GPIOM               GPIOM_BASE
+/** GPION Base Address */
+#define GPION               GPION_BASE
+/** GPIOP Base Address */
+#define GPIOP               GPIOP_BASE
+/** GPIOQ Base Address */
+#define GPIOQ               GPIOQ_BASE
+/**@}*/
+
+/** @defgroup gpio_pin_id GPIO Pin Identifiers
+ * @brief GPIO Pin Identifiers
+@{*/
+/** GPIO Pin 0 Identifier */
+#define GPIO0               (1 << 0)
+/** GPIO Pin 1 Identifier */
+#define GPIO1               (1 << 1)
+/** GPIO Pin 2 Identifier */
+#define GPIO2               (1 << 2)
+/** GPIO Pin 3 Identifier */
+#define GPIO3               (1 << 3)
+/** GPIO Pin 4 Identifier */
+#define GPIO4               (1 << 4)
+/** GPIO Pin 5 Identifier */
+#define GPIO5               (1 << 5)
+/** GPIO Pin 6 Identifier */
+#define GPIO6               (1 << 6)
+/** GPIO Pin 7 Identifier */
+#define GPIO7               (1 << 7)
+/** GPIO All Pins Identifier */
+#define GPIO_ALL            (0xFF)
+/**@}*/
+
+/** @defgroup gpio_af_id GPIO Alternate Functions Identifiers
+ * @brief GPIO Alternate Functions Identifiers
+@{*/
+/** GPIO Alternate Function 1 Identifier */
+#define GPIO_AF1            0x1
+/** GPIO Alternate Function 2 Identifier */
+#define GPIO_AF2            0x2
+/** GPIO Alternate Function 3 Identifier */
+#define GPIO_AF3            0x3
+/** GPIO Alternate Function 4 Identifier */
+#define GPIO_AF4            0x4
+/** GPIO Alternate Function 5 Identifier */
+#define GPIO_AF5            0x5
+/** GPIO Alternate Function 6 Identifier */
+#define GPIO_AF6            0x6
+/** GPIO Alternate Function 7 Identifier */
+#define GPIO_AF7            0x7
+/** GPIO Alternate Function 8 Identifier */
+#define GPIO_AF8            0x8
+/** GPIO Alternate Function 11 Identifier */
+#define GPIO_AF11           0xB
+/** GPIO Alternate Function 13 Identifier */
+#define GPIO_AF13           0xD
+/** GPIO Alternate Function 14 Identifier */
+#define GPIO_AF14           0xE
+/** GPIO Alternate Function 15 Identifier */
+#define GPIO_AF15           0xF
```

Comment:
Can we just drop these? the af number is just ... the af number.

Reply:
Yes, but i think it makes it easier to read.